src/recipes.jl

```diff
@@ -378,109 +378,256 @@ end
 end
 @deps bar shape
 
+
 # ---------------------------------------------------------------------------
 # Histograms
 
-# edges from number of bins
-function calc_edges(v, bins::Integer)
-    vmin, vmax = extrema(v)
-    linspace(vmin, vmax, bins+1)
+_bin_centers(v::AVec) = (v[1:end-1] + v[2:end]) / 2
```

Comment:
> This should use the element-wise plus

Not necessarily, as vector-addition is always element wise - it's different from explicitely defining functions like `sin` for vectors, which is not necessary anymore since Julia has the dot-notation. Of course, in a more complex  expression, `.+` may still have advantages due to fused broadcasting.
In this specific case, though, there was actually a reason why I chose a simple `+`: If you use the current implementation on a `Range` (which is what edges typically are), the result will again be a `Range` - so it's lightweight and fast. If you use `.+`, the result will always be a `Vector`. Not that a bit of memory allocation would be a problem for Plots (performance is obviously limited by other things), but I think it's still elegant.